From 57d7a11191f473b4ff9f105dfe5c70bb17cea1e4 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Wed, 31 May 2017 11:04:52 +0800
Subject: [PATCH 1822/5242] MLK-14989: ASoC: fsl_rpmsg_i2s: enable pm_qos for
 audio

commit  1685304608cad3fb23c47daaf1fe2f8fe66d2928 from
https://source.codeaurora.org/external/imx/linux-imx.git

with "echo 1 > /sys/class/graphics/fb0/blank", and there is no
usb connected on board, the system may enter low power mode,
then audio playback will be failed. use pm_qos to prevent A7
core enter low power mode during audio playback and recording.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_rpmsg_i2s.c |    6 ++++++
 sound/soc/fsl/fsl_rpmsg_i2s.h |    1 +
 2 files changed, 7 insertions(+)

diff --git a/sound/soc/fsl/fsl_rpmsg_i2s.c b/sound/soc/fsl/fsl_rpmsg_i2s.c
index 5a66e90..3cfa3e9 100644
--- a/sound/soc/fsl/fsl_rpmsg_i2s.c
+++ b/sound/soc/fsl/fsl_rpmsg_i2s.c
@@ -178,11 +178,17 @@ static int fsl_rpmsg_i2s_remove(struct platform_device *pdev)
 #ifdef CONFIG_PM
 static int fsl_rpmsg_i2s_runtime_resume(struct device *dev)
 {
+	struct fsl_rpmsg_i2s *rpmsg_i2s = dev_get_drvdata(dev);
+
+	pm_qos_add_request(&rpmsg_i2s->pm_qos_req, PM_QOS_CPU_DMA_LATENCY, 0);
 	return 0;
 }
 
 static int fsl_rpmsg_i2s_runtime_suspend(struct device *dev)
 {
+	struct fsl_rpmsg_i2s *rpmsg_i2s = dev_get_drvdata(dev);
+
+	pm_qos_remove_request(&rpmsg_i2s->pm_qos_req);
 	return 0;
 }
 #endif
diff --git a/sound/soc/fsl/fsl_rpmsg_i2s.h b/sound/soc/fsl/fsl_rpmsg_i2s.h
index 259cb63..be951e9 100644
--- a/sound/soc/fsl/fsl_rpmsg_i2s.h
+++ b/sound/soc/fsl/fsl_rpmsg_i2s.h
@@ -309,6 +309,7 @@ struct i2s_info {
 struct fsl_rpmsg_i2s {
 	struct platform_device *pdev;
 	struct i2s_info        i2s_info;
+	struct pm_qos_request pm_qos_req;
 };
 
 #endif /* __FSL_RPMSG_I2S_H */
-- 
1.7.9.5

